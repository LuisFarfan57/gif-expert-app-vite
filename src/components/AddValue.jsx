import { useState } from 'react'

export const AddValue = ({ onNewValue, placeholder }) => {
  const [newValue, setNewValue] = useState('')

  const handleChange = ({ target }) => {
    setNewValue(target.value)
  }

  const handleKeyPress = ({ key }) => {
    const cleanValue = newValue.trim()
    if (key === 'Enter' && cleanValue.length > 1) {
      onNewValue(cleanValue)
      setNewValue('')
    }
  }

  return (
    <input
      type="text"
      name='newValueInput'
      placeholder={placeholder}
      onChange={handleChange}
      value={newValue}
      onKeyUp={handleKeyPress}
    />
  )
}
