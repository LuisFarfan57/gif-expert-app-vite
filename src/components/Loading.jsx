export const Loading = ({ isLoading }) => {
  return (
    <>
      {isLoading && <h2>Loading...</h2>}
    </>
  )
}
