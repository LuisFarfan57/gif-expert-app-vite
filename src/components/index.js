export * from './AddValue'
export * from './GifGrid'
export * from './GifItem'
export * from './Loading'
