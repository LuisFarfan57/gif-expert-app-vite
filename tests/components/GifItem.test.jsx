import { render } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('GifItem tests', () => {
  const props = {
    title: 'Title',
    url: 'https://www.hola.com/'
  }

  it('renders correctly', () => {
    const { container } = render(<GifItem {...props} />)

    expect(container).toMatchSnapshot()
  })

  it('shows the image and ALT correctly', () => {
    const { getByRole } = render(<GifItem {...props} />)

    const { src, alt } = getByRole('img')
    expect(src).toBe(props.url)
    expect(alt).toBe(props.title)
  })

  it('shows title as text', () => {
    const { getByText } = render(<GifItem {...props} />)

    expect(getByText(props.title)).toBeTruthy()
  })
})
