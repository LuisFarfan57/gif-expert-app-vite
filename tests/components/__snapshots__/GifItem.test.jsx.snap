// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem tests renders correctly 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Title"
      src="https://www.hola.com/"
    />
    <p>
      Title
    </p>
  </div>
</div>
`;
